docs/quickstart.md

```diff
@@ -135,46 +173,76 @@ def cross_entropy(out, Y):
   y = y.reshape(list(Y.shape)+[num_classes])
   y = Tensor(y)
   return out.mul(y).mean()
+
 ```
+
 As we can see in this implementation of cross entropy loss, there are certain operations that tinygrad does not support.
 Namely, operations that are load/store like indexing a tensor with another tensor or assigning a value to a tensor at a certain index.
 Load/store ops are not supported in tinygrad because they add complexity when trying to port to different backends and 90% of the models out there don't use/need them.
 
 For our optimizer we will be using the traditional stochastic gradient descent optimizer with a learning rate of 3e-4.
+
 ```py
 from tinygrad.nn.optim import SGD
 
-opt = SGD([net.l1.weight, net.l2.weight], lr=3e-4)
+#opt = SGD([net.l1.weight, net.l2.weight], lr=3e-4)
+
 ```
+
 We can see that we are passing in the parameters of our neural network to the optimizer.
 This is due to the fact that the optimizer needs to know which parameters to update.
-There is a simpler way to do this just by using `get_parameters(net)` from `tinygrad.nn.optim` which will return a list of all the parameters in the neural network.
 The parameters are just listed out explicitly here for clarity.
+There is a simpler way to do this just by using `get_parameters(net)` from `tinygrad.nn.optim` which will return a list of all the parameters in the neural network like below.
+
+```py
+from tinygrad.nn.optim import get_parameters
+opt = SGD(get_parameters(net), lr=3e-4)
+
+```
 
 Now that we have our network, loss function, and optimizer defined all we are missing is the data to train on!
 There are a couple of dataset loaders in tinygrad located in [/datasets](/datasets).
-We will be using the MNIST dataset loader.
+We will be using the MNIST dataset loader and assign it.
+
 ```py
 from datasets import fetch_mnist
-```
 
-Now we have everything we need to start training our neural network.
-We will be training for 1000 steps with a batch size of 64.
-```py
 X_train, Y_train, X_test, Y_test = fetch_mnist()
 
-for step in range(1000):
+```
+
+We can define a simple function to run our net, it will return our labels and the output of the model.
+
+```py
+def run_net(X,Y,bs):
   # random sample a batch
-  samp = np.random.randint(0, X_train.shape[0], size=(64))
-  batch = Tensor(X_train[samp], requires_grad=False)
+  samp = np.random.randint(0, X.shape[0], size=(bs))
+  batch = Tensor(X[samp], requires_grad=False)
   # get the corresponding labels
-  labels = Y_train[samp]
-
+  labels = Y[samp]
   # forward pass
   out = net(batch)
+  return labels, out
+
+```
+
+Now we have everything we need to start training our neural network.
+We will be training for a target loss of 0.03 with a batch size of 64.
+
+```py
+target_loss = 0.03
+tstep, losst, bs = 0, 1., 64
 
+from tqdm import tqdm
+
+tbar = tqdm(total=0, initial=0, ncols=0, bar_format='Training: {desc}')
```

Comment:
Probably shouldn't use tqdm to keep this guide from depending on more dependencies then just tinygrad.

Reply:
i used tqdm because it is already a requirement from state.py load_state_dict

Reply:
Ah OK yea this is fine then.